src/js/view/controlbar.js

```diff
@@ -233,6 +233,7 @@ define([
             this._model.on('change:captionsList', this.onCaptionsList, this);
             this._model.on('change:captionsIndex', this.onCaptionsIndex, this);
             this._model.on('change:streamType', this.onStreamTypeChange, this);
```

Comment:
Since there is already a listener, call `onDuration` from inside `onStreamTypeChange`. Just make sure to pass duration correctly.